### PR TITLE
Synchronize the two queryist implementations

### PR DESCRIPTION
**Pull request by PavelSafronov**
Synchronize the two queryist implementations so they produce the same results.

- keep the current MySQL iterator that returns everything as a string
- add SqlEngineQueryist, which wraps calls to SqlEngine and returns all data as a string
- SqlEngineQueryist is only going to be used by dolt subcommands, so no one is being broken by this change

## More info about this change

The current Queryist logic flips between 2 implementations:
1. SQLEngine querying - here, we access Dolt directly, and get back properly typed responses. We're also receiving `bool` objects in some cases, like with [the `staged` column  on `dolt_status` table](https://github.com/dolthub/dolt/blob/main/go/libraries/doltcore/sqle/dtables/status_table.go#L49). This mode is used when we are running `dolt foo` from a dolt directory.
2. MySQL querying - we are currently using MysqlRowWrapper, an existing wrapper that considers every column a string and returns strings. This mode is used when we are running `dolt foo` commands against a remote server.

So with the current Queryist logic, if we execute the query `select * from dolt_status`, then the 2nd column will either be `bool` (with SQLEngine) or `string` (with MySQL).
That means that currently Queryist consumers would have to consider and handle both cases.

This PR synchronizes the two flavors of Queryist implementations by always returning strings from dolt. This is a smaller change than the previous version of this PR and does not impact any dolt consumers.

### CODE REVIEW

**Comment by timsehn** [comment]
This is cool work coming out of merging these two paths

File: go/cmd/dolt/commands/mysql_rows_iter.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+	"strings"
+)
+
+// MySqlRowsIter is an iterator over rows returned by MySQL
+type MySqlRowsIter struct {
+	rows        *dsql.Rows
+	types       []sql.Type
+	columnTypes []*dsql.ColumnType
+	schema      sql.Schema
+}
```
Comment (macneale4) [inline go/cmd/dolt/commands/mysql_rows_iter.go:19]: Where is 'types' ever use once the object is constructed? It looks like you never use it, so I wouldn't bother putting it in the struct.

File: go/cmd/dolt/commands/mysql_rows_iter.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+func (b MySqlRowsIter) Close(_ *sql.Context) error {
+	return b.rows.Close()
+}
+
+// Schema returns the schema for the rows returned by this iterator.
+func (b MySqlRowsIter) Schema() sql.Schema {
+	return b.schema
+}
+
+func scanResultRow(results *dsql.Rows, cols []*dsql.ColumnType) (sql.Row, error) {
```
Comment (macneale4) [inline go/cmd/dolt/commands/mysql_rows_iter.go:74]: You're going to need a lot of tests for this to convince me it's right. This feels like a lot of code what we could get wrong.

File: go/cmd/dolt/commands/mysql_rows_iter.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+		scanRow[i] = t
+	}
+}
+
+var typeDefaults = map[string]string{
+	"char":      "char(255)",
+	"binary":    "binary(255)",
+	"varchar":   "varchar(65535)",
+	"varbinary": "varbinary(65535)",
+}
```
Comment (macneale4) [inline go/cmd/dolt/commands/mysql_rows_iter.go:216]: Why so few? What's the thinking on this specific set.

**Review by macneale4 — CHANGES_REQUESTED** [review]
Thanks for going deep on this! This is pretty high risk. Need lots of tests, or I need to be convinced that we have existing tests which cover all of the crazy ways schema's can be different. Given all the logic in these translation laters, we could get this very wrong.

File: go/cmd/dolt/commands/sql_engine_queryist.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+var _ sql.RowIter = (*SqlEngineRowIter)(nil)
+
+func NewSqlEngineRowIter(iter sql.RowIter, schema sql.Schema) *SqlEngineRowIter {
+	newSchema := make(sql.Schema, len(schema))
+	for i, col := range schema {
+		newSchema[i] = &sql.Column{
+			Name:     col.Name,
+			Type:     types.LongText,
```
Comment (macneale4) [inline go/cmd/dolt/commands/sql_engine_queryist.go:94]: I don't understand this. Seems like reduced behavior over what we already have.

**Review by macneale4 — CHANGES_REQUESTED** [review]

**Review by nicktobey — APPROVED** [review]

**Pull request closed without merging**